syntax/diags.go

```diff
@@ -24,31 +24,6 @@ import (
 // A Diagnostic represents a warning or an error to be presented to the user.
 type Diagnostic struct {
 	hcl.Diagnostic
-
-	// Whether the diagnostic has been shown to the user
-	Shown bool
-}
-
-// WithContext adds context without mutating the receiver.
-func (d Diagnostic) WithContext(rng *hcl.Range) *Diagnostic {
-	d.Context = rng
-	return &d
-}
-
-func (d Diagnostic) HCL() *hcl.Diagnostic {
-	return &d.Diagnostic
-}
-
-// Warning creates a new warning-level diagnostic from the given subject, summary, and detail.
-func Warning(rng *hcl.Range, summary, detail string) *Diagnostic {
```

Comment:
Isn't this used in https://github.com/pulumi/esc/blob/d6d07690def94d28b10662009067ae464b159cc6/ast/expr.go#L594-L598?